messages/src/msg_fields/protocols/coordinate_mediation/keylist.rs

```diff
@@ -0,0 +1,91 @@
+use serde::{Deserialize, Serialize};
+use typed_builder::TypedBuilder;
+
+use crate::{decorators::thread::Thread, msg_parts::MsgParts};
+
+pub type Keylist = MsgParts<KeylistContent, KeylistDecorators>;
+
+#[derive(Clone, Debug, Deserialize, Serialize, Default, PartialEq, TypedBuilder)]
+pub struct KeylistContent {
+    pub keys: Vec<KeylistItem>,
+    #[builder(default, setter(strip_option))]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pagination: Option<KeylistPagination>,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, Default, PartialEq, TypedBuilder)]
+pub struct KeylistItem {
+    pub recipient_key: String,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, Default, PartialEq, TypedBuilder)]
+pub struct KeylistPagination {
+    #[builder(default, setter(strip_option))]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    count: Option<u64>,
```

Comment:
while `pagination` field itself is (correctly) optional within `KeylistContent`, I don't think these individual fields are meant to be optional. If mediator decides to provide the pagination information, I think it should be complete. Also don't see indication of otherwise in the rfc